src/witan/gateway/handler.clj

```diff
@@ -112,6 +112,26 @@
                          :kixi.comms.query/results results}))))
 
 ;;
+;;Add partition-key finder here!
+;;exception if there's isn't one
+
+(defn uuid
+  [& _]
+  (str (java.util.UUID/randomUUID)))
+
+(def command-key->partition-key-fn
+  {:kixi.datastore.filestore/create-upload-link uuid
+   :kixi.datastore.schema/create uuid
+   :kixi.datastore.filestore/create-file-metadata :kixi.datastore.metadatastore/id
+   :kixi.datastore.bundles/create-datapack :kixi.datastore.metadatastore/id
+   :kixi.datastore.metadata/sharing-change :kixi.datastore.metadatastore/id
+   :kixi.datastore.metadata/update :kixi.datastore.metadatastore/id})
+
+(defn partition-key-fn
+  [key]
+  (or (key command-key->partition-key-fn)
+      (when (= "test" (namespace key)) uuid)
```

Comment:
Obviously, this is not great, but it helps the integration tests.

Alternatives would be to change the integration tests to 'real' commands or make them in-process so we can use redefs for this function.